src/Makefile

```diff
@@ -79,6 +79,16 @@ $(OUTPUT)/retsnoop.skel.h: $(OUTPUT)/mass_attach.bpf.o
 $(OUTPUT)/retsnoop.o: $(OUTPUT)/retsnoop.skel.h
 $(OUTPUT)/mass_attacher.o: $(OUTPUT)/retsnoop.skel.h $(OUTPUT)/calib_feat.skel.h
 
+../sidecar/target/release/addr2line::
+	$(call msg,CARGO,addr2line)
+	$(Q)cd ../sidecar; cargo build --release
+
+../tools/addr2line: ../sidecar/target/release/addr2line
+	$(call msg,CP,addr2line)
+	$(Q)cp $< $@
+	$(call msg,LLVM_STRIP,addr2line)
+	$(Q)$(LLVM_STRIP) $@
+
 $(OUTPUT)/addr2line.embed.o: ../tools/addr2line
```

Comment:
we can probably just directly depend on ../sidecar/target/release/addr2line and remove the need to copy binary into tools/ (where it would also need to be .gitignore'd). Please also add sidecar/target into .gitignore.

Reply:
This causes two issues.
 1. We need to check if the binary of addr2line is updated by our-self to strip it.
 2. The symbols created by LD are changed because of the change of the path of the input binary.
 
So, the script touch a timestamp file to check if the binary is updated for the item 1.
For the item 2, define additional symbols as aliases generated by LD itself. Link script is a bit overkilled even it is a comprehensive solution.